app/models/text_field_type.rb

```diff
@@ -37,7 +37,7 @@ def text_length
 
   def text_unique
     unless field.field_items.jsonb_contains(:data, text: text).empty?
-      errors.add(:text, "#{field.name} Must be unique")
+      errors.add(:text, "#{field.name} Must be unique") if field_item.content_item_id == nil
```

Comment:
Additionally, I'd make it the first conditional to evaluate, since it's less expensive of a check than the `jsonb_contains` invocation. Ruby supports short-circuit evaluation of expressions when using boolean operators.